contrib_bots/lib/xkcd.py

```diff
@@ -0,0 +1,139 @@
+        ))
+
+
+class XkcdBotCommand(object):
+    LATEST = 0
+    RANDOM = 1
+    COMIC_ID = 2
+
+
+class XkcdNotFoundError(Exception):
+    pass
+
+
+class XkcdServerError(Exception):
+    pass
+
+
+def get_xkcd_bot_response(message):
+    original_content = message['content'].strip()
+    cropped = original_content[len('@xkcd '):]
+    command = cropped.strip()
+
+    xkcd_called_without_command = original_content == '@xkcd'
+
+    try:
+        if command == 'help' or xkcd_called_without_command:
+            return ("xkcd bot support these commands:"
+                    "\n* `@xkcd help` to show this help message."
+                    "\n* `@xkcd latest` fetch latest comic from xkcd."
+                    "\n* `@xkcd random` fetch random comic from xkcd."
+                    "\n* `@xkcd <comic id>` fetch comic based on `<comic id>` "
+                    "e.g `@xkcd 1234`.")
+        elif command == 'latest':
+            fetched = fetch_xkcd_query(XkcdBotCommand.LATEST)
+        elif command == 'random':
+            fetched = fetch_xkcd_query(XkcdBotCommand.RANDOM)
+        elif command.isdigit():
+            fetched = fetch_xkcd_query(XkcdBotCommand.COMIC_ID, cropped.strip())
+        else:
+            return ("xkcd bot only support these commands:"
+                    "\n* `@xkcd help` to show this help message."
+                    "\n* `@xkcd latest` fetch latest comic from xkcd."
+                    "\n* `@xkcd random` fetch random comic from xkcd."
+                    "\n* `@xkcd <comic id>` fetch comic based on `<comic id>` "
+                    "e.g `@xkcd 1234`")
+    except (requests.exceptions.ConnectionError, XkcdServerError):
+        logging.exception('Connection error occurred when trying to connect to xkcd server')
+        return 'Sorry, I cannot process your request right now, please try again later!'
+    except XkcdNotFoundError:
+        logging.exception('XKCD server responds 404 when trying to fetch comic with id %s'
+                          % (command))
+        return 'Sorry, likely there is no xkcd comic with id: #%s' % (command)
+    else:
+        return ("#%s: **%s**\n[%s](%s)" % (fetched['num'],
+                                           fetched['title'],
+                                           fetched['alt'],
+                                           fetched['img']))
+
+
+def fetch_xkcd_query(mode, comic_id=None):
+    try:
+        if mode == XkcdBotCommand.LATEST:  # Fetch latest comic.
+            url = LATEST_XKCD_URL
+
+        elif mode == XkcdBotCommand.RANDOM:  # Fetch random comic.
+            latest = requests.get(LATEST_XKCD_URL)
+
+            if latest.status_code != 200:
+                raise XkcdServerError()
+
+            latest_id = latest.json()['num']
+            random_id = randint(1, latest_id)
+            url = XKCD_TEMPLATE_URL % (str(random_id))
+
+        elif mode == XkcdBotCommand.COMIC_ID:  # Fetch specific comic by id number.
```

Comment:
Okay, bot will detect if the argument is digit, it will fetch a comic strip based on the digit given